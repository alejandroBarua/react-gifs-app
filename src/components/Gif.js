import React from 'react';
import PropTypes from 'prop-types';


const Gif = ({title, url}) => {

	const titleShort = title.slice(0,25) + "...";

	return (
		<div className="card animate__animated animate__fadeIn">
			<p>{titleShort}</p>
			<img src={url} alt={title} title={title} />
		</div>
	)
}

Gif.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired 
}

export default Gif;