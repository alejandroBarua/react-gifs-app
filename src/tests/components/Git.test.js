import React from 'react';
import Gif from '../../components/Gif';
import { shallow } from 'enzyme';

describe('tests <Gif />', () => {
	
	const title = "Hello World",
		url ="htpp://localhost:8080/",
		wrapper = shallow(<Gif title={title} url={url}/>);
		
		test('render Component', () => {

			expect(wrapper).toMatchSnapshot();
	})
	
	test('title', () => {
		
		const titleShort = title.slice(0,25) + "...";
		const $p = wrapper.find('p');

		expect($p.text().trim()).toBe(titleShort);
	})

	test('img', () => {
		
		const $img = wrapper.find('img');

		//console.log(img.html());
		//console.log(img.props());
		//console.log(img.prop('src'));

		expect($img.prop('src')).toBe(url);
		expect($img.prop('alt')).toBe(title);

	})

	test('div', () => {
		
		const $div = wrapper.find('div'),
			className = $div.prop('className');

		expect(className.includes("animate__fadeIn")).toBe(true);

	})

})
 