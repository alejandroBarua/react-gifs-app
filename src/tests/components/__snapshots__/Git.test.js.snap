// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <Gif /> render Component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Hello World...
  </p>
  <img
    alt="Hello World"
    src="htpp://localhost:8080/"
    title="Hello World"
  />
</div>
`;
